File: cf484a_bits.cpp
#include <iostream>

using namespace std;
typedef long long ll;

ll l, r, t;

ll bitsCnt1(ll n) {
    ll ans = 0;
    for (; n; n &= n - 1)
        ans++;
    return ans;
}

ll getBits(ll l, ll r) {
    if (bitsCnt1(r + 1) == 1)
        return r;
    ll d = 127;
    for (d = 127; d >= 0; d--)
        if ((1ll << d) & r)
            break;
    if (l <= (1ll << d) - 1)
        return (1ll << d) - 1;
    return (1ll << d) + getBits(l - (1ll << d), r - (1ll << d));
}

int main() {
    cin >> t;
    while (t--) {
        cin >> l >> r;
        cout << getBits(l, r) << endl;
    }
    return 0;
}

File: cf484a_bits.py
def bitCnt1(n):
    ans = 0
    while n:
        ans, n = ans + 1, n & (n - 1)
    return ans


def getBits(l, r):
    if bitCnt1(r + 1) == 1:
        return r
    d = len(bin(r)[2:])
    k = 1 << d - 1
    return k - 1 if l < k else k + getBits(l - k, r - k)


t = int(input())

for _ in range(t):
    l, r = [int(x) for x in input().split()]
    print(getBits(l, r))
